From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E3=84=97=E3=84=A0=CB=8B=20=E3=84=91=E3=84=A7=CB=8A?=
 <tsao-chi@the-lingo.org>
Date: Thu, 2 Apr 2020 11:35:19 +0800
Subject: [PATCH] Never drop items on command "/give"


diff --git a/src/main/java/net/minecraft/server/CommandGive.java b/src/main/java/net/minecraft/server/CommandGive.java
index 1d22c45af884a917e77e02c272fcbae74794200c..7c55c48610c5629c2cfd2ca537c4de82cd3c8137 100644
--- a/src/main/java/net/minecraft/server/CommandGive.java
+++ b/src/main/java/net/minecraft/server/CommandGive.java
@@ -35,6 +35,7 @@ public class CommandGive {
                 boolean flag = entityplayer.inventory.pickup(itemstack);
                 EntityItem entityitem;
 
+                if (true) continue; // Akarin - never drop items on command "/give"
                 if (flag && itemstack.isEmpty()) {
                     itemstack.setCount(1);
                     entityitem = entityplayer.drop(itemstack, false);
