From 826f2acc2c1b671364e785e9a711319124195b83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E3=84=97=E3=84=A0=CB=8B=20=E3=84=91=E3=84=A7=CB=8A?=
 <tsao-chi@the-lingo.org>
Date: Fri, 3 Apr 2020 15:18:38 +0800
Subject: [PATCH] Server Config


diff --git a/src/main/java/co/aikar/timings/TimingsExport.java b/src/main/java/co/aikar/timings/TimingsExport.java
index 93d5a3f9..08e428bd 100644
--- a/src/main/java/co/aikar/timings/TimingsExport.java
+++ b/src/main/java/co/aikar/timings/TimingsExport.java
@@ -206,7 +206,8 @@ class TimingsExport extends Thread {
         parent.put("config", createObject(
             pair("spigot", mapAsJSON(Bukkit.spigot().getSpigotConfig(), null)),
             pair("bukkit", mapAsJSON(Bukkit.spigot().getBukkitConfig(), null)),
-            pair("paper", mapAsJSON(Bukkit.spigot().getPaperConfig(), null))
+            pair("paper", mapAsJSON(Bukkit.spigot().getPaperConfig(), null)),
+            pair("akarin", mapAsJSON(Bukkit.spigot().getAkarinConfig(), null)) // Akarin - Server Config
         ));
 
         new TimingsExport(listeners, parent, history).start();
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 9ceaac0e..7444b415 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1386,6 +1386,14 @@ public interface Server extends PluginMessageRecipient {
         }
         // Paper end
 
+        // Akarin Start - Server Config
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getAkarinConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Akarin End - Server Config
+
         /**
          * Sends the component to the player
          *
-- 
2.20.1

