From 72ce2ddfec3ddfae14baf3cff08e3182aab14cf0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E3=84=97=E3=84=A0=CB=8B=20=E3=84=91=E3=84=A7=CB=8A?=
 <tsao-chi@the-lingo.org>
Date: Thu, 2 Apr 2020 14:03:29 +0800
Subject: [PATCH] Disable Snooper


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 7a74a0a53..c275944c0 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1169,6 +1169,8 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
         //} // Paper
 
         //this.methodProfiler.enter("snooper"); // Akarin - remove caller
+        // Akarin - Disable Snooper
+        /*
         if (((DedicatedServer) this).getDedicatedServerProperties().snooperEnabled && !this.snooper.d() && this.ticks > 100) { // Spigot
             this.snooper.a();
         }
@@ -1176,6 +1178,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
         if (((DedicatedServer) this).getDedicatedServerProperties().snooperEnabled && this.ticks % 6000 == 0) { // Spigot
             this.snooper.b();
         }
+        */
 
         //this.methodProfiler.exit(); // Akarin - remove caller
         //this.methodProfiler.enter("tallying"); // Akarin - remove caller
-- 
2.25.1.windows.1

